### ISSUE TITLE
Please help me to understand the code

### ISSUE THREAD

**Issue by RahulPhpDev** [aurorascharff/next14-message-box#2]
Hi, I please help me understand this.

In **optimistic-retry** branch , you have used onSubmit={onSubmit} action={submitMessageAction} in form, May I know why we required both if both have same job to submit the form exprct one is client side and another one is server-side-render

**Comment by aurorascharff** [comment]
Yes! So it's not actually required, but I leave the action={} as a fallback to progressively enhance the form. When the js is attached to the form on the client, only the onSubmit will run because it's preventing the default behavior.
